jme3-effects/src/main/java/com/jme3/post/filters/ContrastAdjustmentFilter.java

```diff
@@ -0,0 +1,402 @@
+     * @return this filter instance, for chaining
+     */
+    public ContrastAdjustmentFilter setLowerLimit(float level) {
+        this.lowerLimit = level;
+        if (material != null) {
+            material.setFloat("lowerLimit", lowerLimit);
+        }
+        return this;
+    }
+
+    /**
+     * Returns the lower limit of the input range.
+     * Default value = 0.0.
+     *
+     * @return the lower limit
+     */
+    public float getInputRangeLowerLimit() {
+        return lowerLimit;
+    }
+
+    /**
+     * Returns the upper limit of the input range.
+     * Default value = 1.0.
+     *
+     * @return the upper limit
+     */
+    public float getInputRangeUpperLimit() {
+        return upperLimit;
+    }
+
+    /**
+     * Adjusts the scales of different channels.
+     * Default values = 1.0.
+     *
+     * @param redChannelScale   the red channel scale.
+     * @param greenChannelScale the green channel scale.
+     * @param blueChannelScale  the blue channel scale.
+     * @return this filter instance for a chain call.
+     */
+    public ContrastAdjustmentFilter setScales(float redChannelScale, float greenChannelScale,
+            float blueChannelScale) {
+        setRedScale(redChannelScale);
+        setGreenScale(greenChannelScale);
+        setBlueScale(blueChannelScale);
+
+        return this;
+    }
+
+    /**
+     * Sets the output scale factor for the red channel.
+     *
+     * @param factor the desired scale factor (default=1)
+     * @return this filter instance, for chaining
+     */
+    public ContrastAdjustmentFilter setRedScale(float factor) {
+        this.redChannelScale = factor;
+        if (material != null) {
+            material.setFloat("redChannelScale", redChannelScale);
+        }
+        return this;
+    }
+
+    /**
+     * Sets the output scale factor for the green channel.
+     *
+     * @param factor the desired scale factor (default=1)
+     * @return this filter instance, for chaining
+     */
+    public ContrastAdjustmentFilter setGreenScale(float factor) {
+        this.greenChannelScale = factor;
+        if (material != null) {
+            material.setFloat("greenChannelScale", greenChannelScale);
+        }
+        return this;
+    }
+
+    /**
+     * Sets the output scale factor for the blue channel.
+     *
+     * @param factor the desired scale factor (default=1)
+     * @return this filter instance, for chaining
+     */
+    public ContrastAdjustmentFilter setBlueScale(float factor) {
+        this.blueChannelScale = factor;
+        if (material != null) {
+            material.setFloat("blueChannelScale", blueChannelScale);
+        }
+        return this;
+    }
+
+    /**
+     * Retrieves the value of the red channel scale that's applied on the final pass.
+     * Default value = 1.0.
+     *
+     * @return the scale of the red channel.
+     */
+    public float getRedChannelScale() {
```

Comment:
Could you please standardize the getter and setter methods of the variables? for example `getRedChannelScale` and `setRedScale` -> could you change to get/set RedScale, get/set GreenScale, get/set BlueScale ?

Reply:
Done.